scripts/build/TestImpactAnalysis/storage_query_tool_driver.py

```diff
@@ -0,0 +1,149 @@
+#
+# Copyright (c) Contributors to the Open 3D Engine Project.
+# For complete copyright and license terms please see the LICENSE at the root of this distribution.
+#
+# SPDX-License-Identifier: Apache-2.0 OR MIT
+#
+#
+
+import argparse
+import pathlib
+from tiaf_logger import get_logger
+from s3_storage_query_tool import S3StorageQueryTool
+from local_storage_query_tool import LocalStorageQueryTool
+logger = get_logger(__file__)
+
+
+def parse_args():
+    def valid_file_path(value):
+        if pathlib.Path(value).is_file():
+            return value
+        else:
+            raise FileNotFoundError(value)
+
+    def valid_path(value):
+        if pathlib.Path(value).is_dir():
+            return value
+        else:
+            raise FileNotFoundError(value)
+
+    parser = argparse.ArgumentParser()
+
+    parser.add_argument(
+        '--local',
```

Comment:
I think we can simplify the args like so:

* `--s3-bucket <bucket_name>`: if specified, will use S3 bucket storage, otherwise local storage. The `search-in` arg then handles the path for both local and S3 storage.
* The CRUD operations are mutually exclusive so we could either have an arg that has the mutally exclusive choices or an `action` arg that you can specify one of the mutually exclusive actions to perform.

We can discuss this offline before you amend the PR with these changes.